File: 프로그래머스/0/120880. 특이한 정렬/특이한 정렬.swift
import Foundation

func solution(_ numlist:[Int], _ n:Int) -> [Int] {
    // 튜플 배열로 변환
    let tupleArray = numlist.map { (abs($0 - n), $0) }
    
    // 정렬
    let sortedArray = tupleArray.sorted {
        if $0.0 != $1.0 {
            return $0.0 < $1.0
        } else {
            return $0.1 > $1.1
        }
    }
    
    // 원래 숫자만 추출하여 반환
    return sortedArray.map { $0.1 }
}

File: 프로그래머스/0/120880. 특이한 정렬/특이한 정렬.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

bool compare(const pair<int, int> &left, const pair<int, int> &right) {
    if (left.first != right.first) {
        return left < right;
    } else {
        return left.second > right.second;
    }
}

vector<int> solution(vector<int> numlist, int n) {
    vector<int> answer;
    vector<pair<int, int>> v;

    for (int num : numlist) {
        v.push_back({abs(n - num), num});
    }

    sort(v.begin(), v.end(), compare);

    for(auto pair : v) {
        answer.push_back(pair.second); 
    }

    for(auto num : answer) {
        cout << num << endl;
    }

    return answer;
}